File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    let answer = [0, 0]
    for (let i = 0; i < words.length; i++) {
        let word = words[i]
        let p = i % n + 1
        let turn = Math.ceil((i + 1) / n)
        if (i > 0) {
            let last = words[i - 1].split("").pop()
            if (i > words.indexOf(word) || words[i][0] !== last) {
                answer = [p, turn]
                break
            }
        }
    }
    return answer
}

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    seen_words = set()
    for i, word in enumerate(words):
        player_num = (i % n) + 1
        turn_num = (i // n) + 1
        if i > 0:
            last_char_of_previous_word = words[i - 1][-1]
            if word[0] != last_char_of_previous_word:
                return [player_num, turn_num]
        if word in seen_words:
            return [player_num, turn_num]
        seen_words.add(word)     
    return [0, 0]